library/iFixit/Matryoshka/Memcached.php

```diff
@@ -60,14 +60,34 @@ public function decrement($key, $amount = 1, $expiration = 0) {
        $expiration);
    }
 
+   /**
+    * @template T
+    * @return T|self::MISS
+    */
    public function get($key) {
+      /** @var T|false */
       $value = $this->memcached->get($key);
+      $result = $this->memcached->getResultCode();
+
+      if ($this->isSuccess($result)) {
+         /** @var T */
```

Comment:
If you *store* false... does memcache later return null with a `MISS` result code? or does it still return `false` with a `SUCCESS` code?

I feel like we should have a test around this.

Reply:
RIght now `false` will be returned if that was cached. A miss is only considering the result code, not the stored value. I.E.yes, it is possible to store false in memcache, ask for it, and have it be a successful call

Reply:
>I feel like we should have a test around this.

We already do 🎉